src_c/line.c

```diff
@@ -389,6 +389,72 @@ pg_line_colliderect(pgLineObject *self, PyObject *args)
     return PyBool_FromLong(pgCollision_RectLine(rect, &self->line));
 }
 
+static PyObject *
+pg_line_move(pgLineObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    double Dx = 0, Dy = 0;
+
+    if (nargs == 1) {
+        if (!pg_TwoDoublesFromObj(args[0], &Dx, &Dy)) {
+            goto error;
+        }
+    }
+    else if (nargs == 2) {
+        if (!pg_DoubleFromObj(args[0], &Dx) ||
+            !pg_DoubleFromObj(args[1], &Dy)) {
+            goto error;
+        }
+    }
+    else {
+        goto error;
+    }
+
+    pgLineObject *ret =
+        _pg_line_subtype_new4(Py_TYPE(self), self->line.x1, self->line.y1,
+                              self->line.x2, self->line.y2);
+
+    ret->line.x1 += Dx;
+    ret->line.y1 += Dy;
+    ret->line.x2 += Dx;
+    ret->line.y2 += Dy;
+
+    return ret;
```

Comment:
Right now this code has the wrong types: `_pg_line_subtype_new4` returns a `PyObject *`.
Just do this code fragment in a single line please: 
```suggestion
    return pgLine_New4(self->line.x1 + Dx, self->line.y1 + Dy,
                       self->line.x2 + Dx, self->line.y2 + Dy);
```

Reply:
This will have right types, be more readable and make more sense.

Reply:
this wouldnt work as we need to return the correct subtype

Reply:
although something like this is preferable
```py
return _pg_line_subtype_new4(Py_TYPE(self), self->line.x1 + Dx, self->line.y1 + Dy, self->line.x2 + Dx, self->line.y2 + Dy);
```